### PR TITLE
Fixed #35841 -- Re-enabled support for DB-IP databases in GeoIP2.

### PR DESCRIPTION
**Pull request by ngnpope**
#### Trac ticket number
ticket-35841

#### Branch description
Support for DB-IP Lite databases added in ticket-33442 has regressed. This pull request avoids using `str.endswith()` in favour of a simple `in` check again, but now encapsulated in `.is_city` and `.is_country` properties.

In addition, a new `SUPPORTED_DATABASE_TYPES` global set of supported databases is added which provides a place that can be monkey patched should other database types be required as long as they are broadly compatible with the structure of data provided by MaxMind and DB-IP.

I have managed to create a cut-down version of the DB-IP Lite databases as they do not provide sample/test databases themselves as MaxMind do. This took quite some effort 😓

#### Checklist
- [x] This PR targets the `main` branch. 
- [x] The commit message is written in past tense, mentions the ticket number, and ends with a period.
- [x] I have checked the "Has patch" ticket flag in the Trac system.
- [x] I have added or updated relevant tests.
- [x] I have added or updated relevant docs, including release notes if applicable.
- [x] I have attached screenshots in both light and dark modes for any UI changes.

### CODE REVIEW

**Comment by nessita** [comment]
@HomingHamster could you please test this branch and confirm if this restores the support for DB-IP Lite databases?

**Comment by nessita** [comment]
> Strange that the test suite seems to be stuck, but apart from that, the patch looks very good. Thanks a lot!

Jenkins had a lock on the git repo, I wiped workspaces and I think CI is unblocked. I have pushed a small commit rewording to re-trigger the full run.

File: tests/gis_tests/test_geoip2.py [hunk 1]
```diff
@@ -230,6 +219,30 @@ class GeoIP2Test(GeoLite2Test):
...
+    GEOIP_CITY="dbip-city-lite-test.mmdb",
+    GEOIP_COUNTRY="dbip-country-lite-test.mmdb",
+)
+class DBIPLiteTest(GeoLite2Test):
+    """DB-IP Lite databases are supported."""
+
+    @classproperty
+    def expected_city(cls):
+        # Override result where values differ to other databases.
+        return super().expected_city | {
```
Comment (nessita) [inline tests/gis_tests/test_geoip2.py:233]: Could this just be...?
```suggestion
    expected_city = GeoLite2Test.expected_city | {
```

Comment (ngnpope) [reply tests/gis_tests/test_geoip2.py:233]: Yeah, it could.

**Review by nessita — APPROVED** [review]
Thank you @ngnpope, this looks great, I will push a small fix to the release note to resolve the conflict and use the new ticket number.

**Pull request merged**